multiversx_sdk/abi/managed_decimal_value.py

```diff
@@ -46,53 +46,48 @@ def decode_top_level(self, data: bytes):
             self.scale = 0
             return
 
-        biguint = BigUIntValue()
+        value = BigUIntValue()
         scale = U32Value()
 
         if self.is_variable:
             # read biguint value length in bytes
-            big_uint_size = self._unsigned_from_bytes(data[:U32_SIZE_IN_BYTES])
+            value_length = self._unsigned_from_bytes(data[:U32_SIZE_IN_BYTES])
 
             # remove biguint length; data is only biguint value and scale
             data = data[U32_SIZE_IN_BYTES:]
 
             # read biguint value
-            biguint.decode_top_level(data[:big_uint_size])
+            value.decode_top_level(data[:value_length])
 
             # remove biguintvalue; data contains only scale
-            data = data[big_uint_size:]
+            data = data[value_length:]
 
             # read scale
             scale.decode_top_level(data)
             self.scale = scale.get_payload()
         else:
-            biguint.decode_top_level(data)
+            value.decode_top_level(data)
 
-        self.value = self._convert_to_decimal(biguint.get_payload())
+        self.value = self._convert_to_decimal(value.get_payload())
 
     def decode_nested(self, reader: io.BytesIO):
         length = self._unsigned_from_bytes(read_bytes_exactly(reader, U32_SIZE_IN_BYTES))
         payload = read_bytes_exactly(reader, length)
         self.decode_top_level(payload)
 
     def to_string(self) -> str:
-        value_str = str(self._convert_value_to_int())
-        if self.scale == 0:
-            return value_str
-        if len(value_str) <= self.scale:
-            # If the value is smaller than the scale, prepend zeros
-            value_str = "0" * (self.scale - len(value_str) + 1) + value_str
-        return f"{value_str[:-self.scale]}.{value_str[-self.scale:]}"
+        scaled_value = self._convert_value_to_int()
+        return f"{scaled_value / (10 ** self.scale):.{self.scale}f}"
 
     def get_precision(self) -> int:
-        return len(str(self._convert_value_to_int()).lstrip("0"))
+        value_str = f"{self.value:.{self.scale}f}"
```

Comment:
A double check is necessary here, as well.

Reply:
This should be alright, as well. It behaves like the JS implementation.